crates/typst/src/math/fragment.rs

```diff
@@ -408,9 +410,16 @@ impl VariantFragment {
     /// Vertically adjust the fragment's frame so that it is centered
     /// on the axis.
     pub fn center_on_axis(&mut self, ctx: &MathContext) {
+        self.align_on_axis(ctx, VAlignment::Horizon)
+    }
+
+    /// Vertically adjust the fragment's frame so that it is aligned
+    /// to the given alignment on the axis.
+    pub fn align_on_axis(&mut self, ctx: &MathContext, align: VAlignment) {
         let h = self.frame.height();
         let axis = ctx.constants.axis_height().scaled(ctx, self.font_size);
-        self.frame.set_baseline(h / 2.0 + axis);
+        let align = (2 - align as u8) as f64;
+        self.frame.set_baseline((h / 2.0 + axis) * align);
```

Comment:
can this use `align.position(..)`? The `(2 - align as u8)` is too clever for my liking and also creates an implicit dependence on the order of the enum.

Reply:
This hasn’t been defined for `VAlignment` yet, but I’ve done this.